oioswift/common/middleware/autocontainerbase.py

```diff
@@ -202,6 +206,21 @@ def check_container_obj(alternative):
             env_modifier=modify_copy_from,
             alt_checker=check_container_obj)
 
+    def _skip_metadata(self, env):
+        if not self.skip_metadata:
+            return
+
+        req = Request(env)
+        # don't fake obj metadata
+        account, container, _ = self._extract_path(req.path_info)
+        req.environ['swift.infocache'] = {}
+        req.environ['swift.infocache']['account/%s' % account] = \
+            headers_to_account_info({}, 0)
```

Comment:
The "standard" recipe is to use `req.environ.setdefault('swift.infocache', {})` to create the "infocache" only if it did not exist already, and the use `get_cache_key(account, [...])` (from `swift.proxy.controllers.base`) to build infocache keys.